web/src/peer_data.ts

```diff
@@ -158,3 +158,22 @@ export function is_user_subscribed(stream_id: number, user_id: number): boolean
     const subscribers = get_user_set(stream_id);
     return subscribers.has(user_id);
 }
+
+export function count_guest_visible_users(stream_ids: number[]): number {
+    const valid_subscribers = new LazySet([]);
+
+    for (const stream_id of stream_ids) {
+        const subscribers = get_user_set(stream_id);
+
+        for (const user_id of subscribers.keys()) {
+            if (
+                people.maybe_get_user_by_id(user_id) &&
+                people.is_person_active(user_id) &&
```

Comment:
I am actually not sure about this check well. The subscribers list should only contain active users. We have this check in `get_subscriber_count` but not in `get_subscribers`.